component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/util/TOTPAuthenticatorCredentials.java

```diff
@@ -345,4 +351,48 @@ private boolean authorize(String secretKey, int verificationCode, long time) {
 		// Checking the validation code using the current UNIX time.
 		return checkCode(secretKey, verificationCode, time, this.config.getWindowSize());
 	}
+
+	/**
+	 * Validate verification code with the secret key and store the secret key in secret key claim.
+	 *
+	 * @param secretKey        Secret Key
+	 * @param verificationCode Verification code
+	 * @param username         Username
+	 * @return Whether the verification code is valid or not.
+	 */
+	public boolean authorizeAndStoreSecret(String secretKey, int verificationCode, String username) {
+
+		if (authorize(secretKey, verificationCode, new Date().getTime())) {
+			storeSecretKey(secretKey, username);
+			return true;
+		} else {
+			return false;
+		}
+	}
+
+	private void storeSecretKey(String secretKey, String username) {
+
+		Map<String, String> userClaims = new HashMap<>();
+		Map<String, String> removedUserClaims = new HashMap<>();
```

Comment:
Do we need a variable to keep remove user claims? looks like it is not used

Reply:
Fixed https://github.com/wso2-extensions/identity-outbound-auth-totp/pull/147/commits/875733be2eb959dee4cc57910325d11cee69111e